### PR TITLE
attempting to get completely installed devEnvironment

### PR DESCRIPTION
**Pull request by tommysullivan**

### CODE REVIEW

**Comment by tommysullivan** [comment]
hey @bob-bins @aashreya shall we commit this? although there are some loose ends, there is also some value here and some cleanup / improvements of our bash scripts that i would like to make use of here and even for other projects

we could spend more time on it but i was hoping to get it merged as is. what do you think?

File: master [hunk 1]
```diff
@@ -0,0 +1 @@
+<html><body>You are being <a href="https://codeload.github.com/Homebrew/brew/legacy.tar.gz/master">redirected</a>.</body></html>
```
Comment (bob-bins) [inline master]: what's this for

File: heartpoints.sh [hunk 2]
```diff
@@ -500,89 +505,166 @@ heartpoints_testUntilSuccess() { local timeoutSeconds=$1; local interval=$2; loc
...
-heartpoints_minikubeBuild() { local taggedImageName=$1; local shaToReportInHttpHeaders=$2
+hp_minikubeBuild_help() { echo "<taggedImageName> using minikube's docker daemon, build image and tag with minikube metadata"; }
+hp_minikubeBuild() { local taggedImageName=$1; local shaToReportInHttpHeaders=$2
     requiredParameter "taggedImageName" "${taggedImageName}"
     requiredParameter "shaToReportInHttpHeaders" "${shaToReportInHttpHeaders}"
-    heartpoints_pointToAndRunMinikubeDockerDaemon
-    heartpoints_pointToAndRunMinikubeDockerDaemon
-    heartpoints_buildAndTagImage "${taggedImageName}" "${shaToReportInHttpHeaders}"
+    hp_pointToAndRunMinikubeDockerDaemon
+    hp_pointToAndRunMinikubeDockerDaemon
```
Comment (bob-bins) [inline heartpoints.sh]: why is this here twice

File: heartpoints.sh [hunk 3]
```diff
@@ -500,89 +505,166 @@ heartpoints_testUntilSuccess() { local timeoutSeconds=$1; local interval=$2; loc
...
+    hp_buildAndTagImage "${taggedImageName}" "${shaToReportInHttpHeaders}"
+}
+
+hp_minikubeDestroyEnvironment_help() { echo "if minikube dev environment is running, destroys it"; }
+hp_minikubeDestroyEnvironment() {
+    hp_minikube delete
```
Comment (bob-bins) [inline heartpoints.sh]: i'd personally rather have a function that calls the locally-installed minikube cli and require the user to have some knowledge of how to use the cli than to have wrapper functions such as this

File: heartpoints.sh [hunk 4]
```diff
@@ -610,23 +692,58 @@ gcloud_configure() {
...
+
+homebrew_tar_download_log_path() {
+    hp_log_path "homebrew_tar_download.log"
+}
+
+brew_install_log_path() {
+    hp_log_path "brew_install_brew_itself.log"
+}
+
+brew_install_brew_itself() {
```
Comment (bob-bins) [inline heartpoints.sh]: perhaps we should be explicit about which functions are installed locally within this repo vs globally

File: heartpoints.sh [hunk 5]
```diff
@@ -639,34 +756,47 @@ file_does_not_exist() { local possibleFilePath=$1
...
-    virtualbox_install
-    if command_does_not_exist minikube; then
-        curl -Lo minikube https://storage.googleapis.com/minikube/releases/v0.32.0/minikube-darwin-amd64 && chmod +x minikube && sudo cp minikube /usr/local/bin/ && rm minikube
-    fi
-}
+# kubectl_install() {
+#     brew_install "kubernetes-cli"
+# }
+
+# minikube_install() {
```
Comment (bob-bins) [inline heartpoints.sh]: should this (and `kubectl_install`) remain commented out?

File: heartpoints.sh [hunk 6]
```diff
@@ -500,89 +505,166 @@ heartpoints_testUntilSuccess() { local timeoutSeconds=$1; local interval=$2; loc
...
+        hp_brew install "${packageName}" > "$(hp_log_path "brewPackageInstall_${packageName}.log")" 2>&1
+    fi
+    "${brewPackagePath}" "${@:2}"
+}
+
+brew_package_path() { local packageName=$1;
+    echo "$(hp_brew --prefix "${packageName}")/bin/${packageName}"
+}
+
+hp_slack() {
```
Comment (bob-bins) [inline heartpoints.sh]: Slack isn't technically a requirement, and this function isn't referenced anywhere (including the README). I think it can be deleted since we want to keep this `.sh` file as small as we can

File: master (hunk 1 above)
Comment (tommysullivan) [reply master]: removed

File: heartpoints.sh (hunk 3 above)
Comment (tommysullivan) [reply heartpoints.sh]: i think there's two points you could be making

- rather than have a source controlled way to call the minikube cli (or any other cli we depend on), count on a separately installed minikube cli already existing
- specifically for `hp minikubeDestroyEnvironment` you prefer that user would just use the choice minikube cli to discover and subsequently run the `delete` command

can you confirm / clarify around these :)

File: heartpoints.sh (hunk 6 above)
Comment (tommysullivan) [reply heartpoints.sh]: I can add it to the readme, because I am trying to codify and automated everything needed to work with us, including setting up slack. will also add our slack url to the output. i'll add something called "setupDevEnvironment" that can generally include this and other items and then refer to the broad command in the readme.

File: heartpoints.sh (hunk 4 above)
Comment (tommysullivan) [reply heartpoints.sh]: can you elaborate on this? :)

File: heartpoints.sh (hunk 5 above)
Comment (tommysullivan) [reply heartpoints.sh]: i think there may be something to still do here let me check

**Comment by tommysullivan** [comment]
@bob-bins tried to address what i could while you were napping, hopefully we can get this merged!

File: heartpoints.sh (hunk 3 above)
Comment (bob-bins) [reply heartpoints.sh]: the first point is not one that I'm making (though i'd be fine with that as well). it's just the second point

File: heartpoints.sh (hunk 4 above)
Comment (bob-bins) [reply heartpoints.sh]: some of the brew installations in this PR  install into a directory within this repository. and others, like Virtualbox, are installed globally for that user/computer. It would be great to have a comment or convention to differentiate this
